Realm.Shared/Realm.cs

```diff
@@ -115,7 +115,30 @@ internal static Realm GetInstance(RealmConfiguration config, RealmSchema schema)
                 }
             }
 
-            var srPtr = srHandle.Open(config.DatabasePath, config.ReadOnly, false, config.EncryptionKey, schema, config.ShouldDeleteIfMigrationNeeded, config.SchemaVersion);
+            var configuration = new Native.Configuration
+            {
+                Path = config.DatabasePath,
+                read_only = config.ReadOnly,
+                delete_if_migration_needed = config.ShouldDeleteIfMigrationNeeded,
+                schema_version = config.SchemaVersion
+            };
+
+            Migration migration = null;
+            if (config.MigrationCallback != null)
+            {
+                migration = new Migration(config, schema);
+                migration.PopulateConfiguration(ref configuration);
+            }
+
+            var srPtr = IntPtr.Zero;
+            try
+            {
+                srPtr = srHandle.Open(configuration, schema, config.EncryptionKey);
+            }
+            catch (ManagedExceptionDuringMigrationException)
+            {
+                throw new AggregateException("Exception occurred in a Realm migration callback. See inner exception for more details.", migration?.MigrationException);
```

Comment:
I think it's more a matter of the `?` operator being redundant. @fealebenpae is there any way for `migration` to be null when this exception is caught?